doc/NAT/SAI-NAT-API.md

```diff
@@ -0,0 +1,443 @@
+In the outbound direction, the SIP is translated in same fashion. Reverse entry should be created in similar manner.
+
+> From Zone ID: 100
+To Zone ID: 200
+VRF: None
+Packet Count: Enable
+Byte Count: Enable
+DNAT Entry
+ExternalEndpoint: 200.0.0.0/24
+InternalEndpoint: 128.17.18.0/24
+
+##### Step 1: Create a Destination NAT Entry object:
+```sh
+
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t subnet_nat_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DESTINATION_NAT;
+
+nat_entry_attr[1].id = SAI_NAT_ENTRY_ATTR_DST_IP;
+nat_entry_attr[1].value.u32 = 128.17.18.0; //example string
+
+nat_entry_attr[2].id = SAI_NAT_ENTRY_ATTR_DST_IP_MASK;
+nat_entry_attr[2].value.u32 = 0xffffff00;
+
+nat_entry_attr[3].id = SAI_NAT_ENTRY_ATTR_TO_ZONE;
+nat_entry_attr[3].value.u32 = 200;
+nat_entry_attr[4].id = SAI_NAT_ENTRY_ATTR_FROM_ZONE;
+nat_entry_attr[4].value.u32 =100;
+
+nat_entry_attr[5].id = SAI_NAT_ENTRY_ATTR_ENABLE_PACKET_COUNT;
+nat_entry_attr[5].value.bool = true;
+
+nat_entry_attr[6].id = SAI_NAT_ENTRY_ATTR_ENABLE_BYTE_COUNT;
+nat_entry_attr[6].value.bool = true;
+
+attr_count = 7;
+
+memset(&subnet_nat_entry, 0, sizeof(nat_entry));
+
+subnet_nat_entry.data.key.dst_ip = 200.0.0.0;
+subnet_nat_entry.data.mask.dst_ip = 0xffffff00;
+
+create_nat_entry(&subnet_nat_entry, attr_count, nat_entry_attr);
+```
+### 5.4 DNAT Pool Configuration 
+For dynamic NAT, translation is setup by the control plane based on the miss. This section describes how a DNAT pool entry is setup to trigger a miss for a given prefix pool. Workflow will create a DNAT pool entry with single attribute of type DNAT_POOL. Packets matching DNAT pool prefix will either result in a hit or a miss based on if the dnat translation is installed in hardware or not.
+
+```sh
+
+sai_attribute_t nat_entry_attr[10];
+nat_entry_t dnat_pool_entry;
+
+nat_entry_attr[0].id = SAI_NAT_ENTRY_ATTR_NAT_TYPE;
+nat_entry_attr[0].value = SAI_NAT_TYPE_DESTINATION_NAT_POOL;
+
+attr_count = 1;
+
+memset(&dnat_pool_entry, 0, sizeof(nat_etnry));
+
+dnat_pool_entry.data.key.dst_ip = 65.55.42.1;
+dnat_pool_entry.data.mask.dst_ip = 0xffffffff;
+
+create_nat_entry(&dnat_pool_entry, attr_count, nat_entry_attr);
+```
+
+## 6.0 NAT Exceptions
+NAT exceptions can be defined using the NAT entry. In that case they will be placed in the SNAT/DNAT table. Operator MUST consider defining the exceptions so as not to collide with other NAT entries.
+An override NAT exception can be created using the ACL rule. A new ACL action is defined in saiacl.h
+
+> SAI_ACL_ENTRY_ATTR_ACTION_NO_NAT
+
+In SAI pipeline if there is a ACL match with action NO_NAT and there is also a match in SNAT/DNAT table then ACL match result always takes precedence.
+
+### 6.1 NAT exception using NAT Entry

```

Comment:
The details in this section seem incorrect. It is not a NAT entry that creates the exceptions, it is an ACL entry. There is no SAI_NAT_TYPE_NO_NAT.

Reply:
Left over from old API. Removed section 6.1